great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -648,3 +650,27 @@ def add_checkpoint(
             checkpoint_config=checkpoint_config, data_context=self  # type: ignore[arg-type]
         )
         return checkpoint
+
+    def _init_site_builder_for_data_docs_site_creation(
+        self, site_name: str, site_config: dict
+    ) -> SiteBuilder:
+        """
+        Note that this explicitly overriding the `AbstractDataContext` helper method called
+        in `self.build_data_docs()`.
+
+        The only difference here is the inclusion of `ge_cloud_mode` in the `runtime_environment`
+        used in `SiteBuilder` instantiation.
+        """
+        site_builder: SiteBuilder = instantiate_class_from_config(
+            config=site_config,
+            runtime_environment={
+                "data_context": self,
+                "root_directory": self.root_directory,
+                "site_name": site_name,
+                "ge_cloud_mode": self.ge_cloud_mode,
+            },
+            config_defaults={
+                "module_name": "great_expectations.render.renderer.site_builder"
+            },
+        )
+        return site_builder
```

Comment:
This is the only piece of code I've introduced in the PR - everything else is a 1-to-1 porting of logic from `BaseDataContext` to `AbstractDataContext`.

Is this the best way to do this? I only want to reference `ge_cloud_mode` in Cloud-specific files; this allows us to do this.

Reply:
Just my 2-cents, I think this **is** my preferred way to handle `ge_cloud_mode` in Cloud-specific files.  If you're concerned about code-duplication, I think your docstring makes it really clear why this method exists. I like it 👍🏼